src/analysis/complex/conformal.lean

```diff
@@ -0,0 +1,181 @@
+/-
+Copyright (c) 2021 Yourong Zang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yourong Zang
+-/
+import analysis.complex.isometry
+import analysis.calculus.conformal
+
+/-!
+# Complex Conformal Maps
+
+We prove the sufficient and necessary conditions for a complex function to be conformal.
+
+## Main results
+
+* `is_complex_or_conj_complex_linear_iff_is_conformal_map`: the main theorem for linear maps.
+* `conformal_at_iff_holomorphic_or_antiholomorphic_at`: the main theorem for complex functions.
+
+## Warning
+
+Antiholomorphic functions such as the complex conjugate are considered as conformal functions in
+this file.
+-/
+
+noncomputable theory
+
+open complex linear_isometry linear_isometry_equiv continuous_linear_map
+     finite_dimensional linear_map
+
+section complex_normed_space
+
+variables {E : Type*} [normed_group E] [normed_space ℝ E] [normed_space ℂ E]
+  [is_scalar_tower ℝ ℂ E] {z : ℂ} {g : ℂ →L[ℝ] E} {f : ℂ → E}
+
+lemma differentiable_at_iff_exists_linear_map (hf : differentiable_at ℝ f z) :
+  differentiable_at ℂ f z ↔ ∃ (g' : ℂ →L[ℂ] E), g'.restrict_scalars ℝ = fderiv ℝ f z :=
+sorry
+
+lemma antiholomorph_at_iff_exists_complex_linear_conj (hf : differentiable_at ℝ f z) :
+  differentiable_at ℂ (f ∘ conj) (conj z) ↔
+  ∃ (g' : ℂ →L[ℂ] E), g'.restrict_scalars ℝ =
+  (fderiv ℝ f z).comp conj_cle.to_continuous_linear_map :=
+sorry
+
+lemma is_conformal_map_of_eq_complex_linear (nonzero : g ≠ 0)
+  (H : ∃ (map : ℂ →L[ℂ] E), map.restrict_scalars ℝ = g) : is_conformal_map g :=
+begin
+  rcases H with ⟨map, h⟩,
+  have minor₀ : ∀ x, g x = x • g 1,
+  { intro x,
+    nth_rewrite 0 ← mul_one x,
+    rw [← smul_eq_mul, ← h, coe_restrict_scalars', map.map_smul], },
+  have minor₁ : ∥g 1∥ ≠ 0,
+  { contrapose! nonzero with w,
+    ext1 x,
+    rw [continuous_linear_map.zero_apply, minor₀ x, norm_eq_zero.mp w, smul_zero], },
+  refine ⟨∥g 1∥, minor₁, _⟩,
+  let li' := ∥g 1∥⁻¹ • g,
+  have key : ∀ (x : ℂ), ∥li' x∥ = ∥x∥,
+  { intros x,
+    simp only [li', continuous_linear_map.smul_apply, minor₀ x,
+               norm_smul, normed_field.norm_inv, norm_norm],
+    field_simp [minor₁], },
+  let li : ℂ →ₗᵢ[ℝ] E := ⟨li', key⟩,
+  have minor₂ : (li : ℂ → E) = li' := rfl,
+  refine ⟨li, _⟩,
+  ext1,
+  simp only [minor₂, li', pi.smul_apply, continuous_linear_map.smul_apply, smul_smul],
+  field_simp [minor₁],
+end
+
+lemma is_conformal_map_of_eq_linear_conj (nonzero : g ≠ 0)
+  (h : ∃ (map : ℂ →L[ℂ] E), map.restrict_scalars ℝ = g.comp conj_cle.to_continuous_linear_map) :
+  is_conformal_map g :=
```

Comment:
I think you want to extract lemmas (1) an isometry satisfies `is_conformal_map`, (2) `conj_cle` satisfies `is_conformal_map`; then this lemma should follow from `is_conformal_map.comp`.